simple_slice_rust.rs

```diff
@@ -0,0 +1,162 @@
+use pyo3::prelude::*;
+use pyo3::types::{PySlice, PyTuple, PyBool, PyDict, PyType};
+use pyo3::exceptions::{PyTypeError, PyValueError};
+use pyo3::PyResult;
+
+#[pyclass(name = "SimpleSliceRust", subclass)]
+struct SimpleSliceRust {
+    _start: isize,
+    _stop: isize,
+    _step: isize,
+    _has_start: bool,
+    _has_stop: bool,
+    _has_step: bool,
```

Comment:
Rust is very good at dealing with this data type. Using Option will result in much cleaner code.
```suggestion
    _start: Option<isize>,
    _stop: Option<isize>,
    _step: Option<isize>,
```